exec/java-exec/src/main/resources/rest/profile/profile.ftl

```diff
@@ -171,7 +188,30 @@ table.sortable thead .sorting_desc { background-image: url("/static/img/black-de
   <#assign queued = queueName != "" && queueName != "-" />
 
   <div class="page-header"></div>
-  <h3>Query Profile : <span style='font-size:85%'>${model.getQueryId()}</span></h3>
+  <h3>Query Profile: <span style='font-size:85%'>${model.getQueryId()}</span>
+  <#if model.getQueryStateDisplayName() == "Running" || model.getQueryStateDisplayName() == "Planning">
```

Comment:
What about other states when query can be cancelled?
STARTING
ENQUEUED
PREPARING

Reply:
I believe a QueryId is not created during STARTING. Will need to check for other states... as I'm not sure what PREPARING does

Reply:
Actually, on second thoughts, I can add STARTING, etc as well. If the query ID doesnt exist, the page wouldn't exist, so we should be fine. I'll add the change.